### PR TITLE
[upgrade] added functionality to handle Properties

### PR DESCRIPTION
**Pull request by Veethahavya**
Solves:
- https://github.com/mfarragher/obsidiantools/issues/39
- https://github.com/mfarragher/obsidiantools/issues/44

Credits:
- GHCP Agent; Sonnet 3.5

### CODE REVIEW

**Review by Veethahavya — COMMENTED** [review]
also added:
- a notebook to illustrate functionality

**Comment by JavierGOrdonnez** [comment]
@mfarragher would be great if this PR could go in (allowing support for frontmatter properties is even more important now with the release of Obsidian Bases)

File: obsidiantools/md_utils.py [hunk 1]
```diff
@@ -509,3 +512,161 @@ def _remove_embedded_file_links_from_text(src_txt: str) -> str:
...
+                      new_list.extend(_get_all_wikilinks_from_source_text(item, remove_aliases=True))
+                  else:
+                      new_list.append(item)
+                  value = new_list
+                if clean_key in ('date', 'due') and all(isinstance(item, str) and len(item) == 10 and item[4] == '-' and item[7] == '-' for item in value):
+                    try:  # Convert to date objects if they match the pattern
+                        value = datetime.datetime.strptime(value, '%Y-%m-%d').date()
+                    except ValueError:
+                        pass
+            props[clean_key] = value
```
Comment (mfarragher) [inline obsidiantools/md_utils.py:594]: This parsing looks like it could be simplified

**Review by mfarragher — APPROVED** [review]
I'm going to merge this into a branch called `dev_with_properties` to do further development.

Examples of tweaks I'll make:
- `.obsidian/types.json` contains types like `datetime`.  That could be used to make the parsing logic more robust and generalise better than it does currently.
- I'll re-organise the files used for tests
- Some of the original tests that check the vault object are failing (they need to account for the new `rich.md` file in this PR).

**Comment by mfarragher** [comment]
@Veethahavya I've reworked the logic so that it is all integrated in `api.py` and parses the date & datetime properties via the `types.json` config file.

However a few edge case tests aren't working now and I'm not sure why, but it might be quicker for you to debug?: issue https://github.com/mfarragher/obsidiantools/issues/56
(possibly to do with aspects like how the overrides work)
